posts/2018-11-02-semirings.md

```diff
@@ -0,0 +1,438 @@
+---
+layout: post
+title: A tale on Semirings
+
+meta:
+  nav: blog
+  author: lukajcb
+  pygments: true
+
+tut:
+  scala: 2.12.7
+  binaryScala: "2.12"
+  scalacOptions:
+    - "-Ypartial-unification"
+    - "-language:higherKinds"
+    - "-language:implicitConversions"
+    - "-language:experimental.macros"
+  dependencies:
+    - com.github.mpilquist::simulacrum:0.14.0
+  plugins:
+    - org.scalamacros:paradise_2.12.7:2.1.0
+---
+
+
+
+*Ever wondered why sum types are called sum types?
+Or maybe you've always wondered why the `<*>` operator uses exactly these symbols?
+And what do these things have to do with Semirings?
+Read this article and find out!*
+
+We all know and use `Monoid`s and `Semigroup`s.
```

Comment:
Minor comment: for folks that don't know or use these, this sentence may drive them off. I suggest rewording with a link to some background info if the reader isn't comfortable with these concepts.

Reply:
Yeah, agreed :+1: